packages/hoppscotch-backend/src/user-collection/user-collection.service.ts

```diff
@@ -1041,9 +1079,60 @@ export class UserCollectionService {
     );
 
     userCollections.forEach((x) =>
-      this.pubsub.publish(`user_coll/${userID}/created`, x),
+      this.pubsub.publish(`user_coll/${userID}/created`, this.cast(x)),
     );
 
     return E.right(true);
   }
+
+  /**
+   * Update a UserCollection
+   *
+   * @param newTitle The new title of collection
+   * @param userCollectionID The Collection Id
+   * @param userID The User UID
+   * @returns An Either of the updated UserCollection
+   */
+  async updateUserCollection(
+    newTitle: string = null,
+    collectionData: string | null = null,
+    userCollectionID: string,
+    userID: string,
+  ) {
+    if (collectionData) {
+      const jsonReq = stringToJson(collectionData);
+      if (E.isLeft(jsonReq)) return E.left(USER_COLL_DATA_INVALID);
+      collectionData = jsonReq.right;
+    }
+
+    if (newTitle != null) {
+      const isTitleValid = isValidLength(newTitle, this.TITLE_LENGTH);
+      if (!isTitleValid) return E.left(USER_COLL_SHORT_TITLE);
+    }
+
+    // Check to see is the collection belongs to the user
+    const isOwner = await this.isOwnerCheck(userCollectionID, userID);
+    if (O.isNone(isOwner)) return E.left(USER_NOT_OWNER);
+
+    try {
+      const updatedUserCollection = await this.prisma.userCollection.update({
+        where: {
+          id: userCollectionID,
+        },
+        data: {
+          data: collectionData ?? undefined,
```

Comment:
If frontend wants to nullify the value of `data`, it will not become `null`

Reply:
Ideally, the frontend will never set the value of `data` as null, it is set it null now for compatability reasons.

Reply:
Ideally the frontend won't set the value of `data` to be `null`, the `data` field is `null` for a compatability thing.